### ISSUE TITLE
React Axe error with Material UI Select

### ISSUE THREAD

**Issue by jpasquers** [dequelabs/react-axe#103]
React Axe produces the following error 

`axe.js:3139 Uncaught (in promise) Error: No elements found for include in page Context
    at validateContext (axe.js:3139)
    at new Context (axe.js:3182)
    at Object.runRules [as _runRules] (axe.js:4012)
    at Object.axe.run (axe.js:4200)
    at requestIdleCallback.timeout (index.js:131)
    at runTasks (index.js:138)`

When a Material UI Select dropdown is opened and then closed. I believe this has to do with the React Portal that Material UI opens, and the fact that it is hidden for a brief time before the element is removed.

**Demo**: https://codesandbox.io/s/elastic-fast-hvo5d
(Note, in this demo the code sits within an iFrame. However the issue also occurs within our code which is outside of an iFrame)

**Comment by jpasquers** [comment]
I have found where I believe the bug to be. What is happening is that componentDidUpdate on my component is being fired, which causes the checkNode to run. However, during the timeout (although this happened when timeout was set to 0, so I'm not positive here), the componentWillUnmount is called, and the component is destroyed. Therefore, when the timeout completes, it will attempt to call Axe on a node that is no longer attached. My solution (which did solve the error when I tried it), was to add a hook in 'componentAfterRender' attached to the componentWillUnmount lifecycle, and then in the resulting method I removed the node from the 'nodes' list. If you think this is a generally good approach I can make a PR of the changes.
